src/ebmc/bdd_engine.cpp

```diff
@@ -566,67 +574,24 @@ Function: bdd_enginet::AGAFp
 
 \*******************************************************************/
 
-void bdd_enginet::AGAFp(propertyt &property)
+void bdd_enginet::check_CTL(propertyt &property)
 {
-  const exprt &sub_expr =
-    to_unary_expr(to_unary_expr(property.normalized_expr).op()).op();
-  BDD p = property2BDD(sub_expr);
-
-  // Start with p, and go backwards until saturation.
-  // AG AFp holds iff this includes all initial states.
-
-  BDD good_states = p;
-  unsigned iteration = 0;
-
-  for(const auto &c : constraints_BDDs)
-    good_states = good_states & c;
-
-  std::size_t peak_bdd_nodes = 0;
-
-  while(true)
-  {
-    iteration++;
-    message.statistics() << "Iteration " << iteration << messaget::eom;
-
-    // make the states be expressed in terms of 'next' variables
-    BDD good_states_next = current_to_next(good_states);
-
-    // now conjoin with transition relation
-    BDD conjunction = good_states_next;
-
-    for(const auto &t : transition_BDDs)
-      conjunction = conjunction & t;
-
-    for(const auto &c : constraints_BDDs)
-      conjunction = conjunction & c;
-
-    // now project away 'next' variables
-    BDD pre_image = project_next(conjunction);
-
-    // compute union
-    BDD set_union = good_states | pre_image;
-
-    // have we saturated?
-    if((set_union == good_states).is_true())
-    {
-      message.progress() << "Fixedpoint reached" << messaget::eom;
-      break;
-    }
-
-    good_states = set_union;
-
-    peak_bdd_nodes = std::max(peak_bdd_nodes, mgr.number_of_nodes());
-  }
+  // For a CTL formula f, calculate the set of states that satisfy
+  // !f, and then check if that contains any initial state.
+  BDD not_f = !CTL(property.normalized_expr);
 
   // Does 'good_states' include all initial states?
-  BDD union_with_initial = good_states;
+  BDD intersection = not_f;
```

Comment:
I believe the comment needs updating (there is no "good_states" object anymore, and it should be "any" instead of "all" I believe).